tsl/test/sql/dist_grant.sql

```diff
@@ -262,7 +262,9 @@ SELECT * FROM create_distributed_hypertable('disttable_role_3', 'time', data_nod
 \set ON_ERROR_STOP 1
 
 RESET ROLE;
-CREATE USER MAPPING FOR :ROLE_3 SERVER data2 OPTIONS (user :'ROLE_3', password :'ROLE_3_PASS');
+-- Create user mapping for ROLE_3, but don't specify user in
+-- options. The "current user" is assumed instead.
```

Comment:
And what is the current user? Isn't it the super user used to connect to this database?

Reply:
Actually not passing user means that `:ROLE_3` has no rights and thus the test fails no.

Reply:
I did see that test does not pass on Windows, so will investigate. But the test passes on other platforms and I don't understand what you mean by "no rights and thus the test fails"? Can you elaborate?

FWIW, the current user is what is returned by `GetUserId()` and (also reflected by) `SELECT current_user;`. AFAIK, user mappings associate no permissions with the user.

Reply:
And the current user is not ROLE_3 to my understanding, thus the mapping doesn't work correctly, I guess. Strange that it behaves differently between Windows and Linux/MacOS.

Reply:
The current user is resolved when the connection is made, not when the mapping is created.

Reply:
Is it possible to show the user mapping, e.g., which remote user will be used in the connection, in the test before setting the user to `:ROLE_3` and after?

Reply:
It is possible to show the contents of `pg_user_mapping`, but that only shows what was added when creating the mapping, which now doesn't include a user. Maybe we could print the connection options somehow, but we don't have a way to do it right now.

Reply:
I think I figured it out. Stupid mistake. I actually never added the user to the connection options.